operators/test/e2e/cmd/run/run.go

```diff
@@ -0,0 +1,561 @@
+						}
+					default:
+					}
+				}
+			case corev1.PodSucceeded:
+				log.Info("Tests completed successfully", "name", newPod.Name)
+				cancelFunc()
+			case corev1.PodFailed:
+				log.Info("Pod is in failed state", "name", newPod.Name)
+				err = errors.New("tests failed")
+				cancelFunc()
+			default:
+				log.Info("Waiting for pod to be ready", "name", newPod.Name, "status", newPod.Status.Phase)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			pod := obj.(*corev1.Pod)
+			log.Info("Pod deleted", "name", pod.Name)
+			cancelFunc()
+		},
+	})
+
+	informer.Run(ctx.Done())
+	return err
+}
+
+func (h *helper) streamTestJobOutput(streamStatus chan<- error, client *kubernetes.Clientset, pod string) {
+	log.Info("Streaming pod logs", "name", pod)
+	defer close(streamStatus)
+	sinceSeconds := int64(30)
+	opts := &corev1.PodLogOptions{
+		Container:    "e2e",
+		Follow:       true,
+		SinceSeconds: &sinceSeconds,
+	}
+
+	req := client.CoreV1().Pods(h.testContext.E2ENamespace).GetLogs(pod, opts).Context(context.Background())
+	stream, err := req.Stream()
+	if err != nil {
+		streamStatus <- err
+		return
+	}
+	defer stream.Close()
+
+	var buffer [logBufferSize]byte
+	if _, err := io.CopyBuffer(os.Stdout, stream, buffer[:]); err != nil {
+		if err == io.EOF {
+			log.Info("Log stream ended")
+			return
+		}
+
+		streamStatus <- err
+	}
+}
+
+func (h *helper) kubectlApplyTemplate(templatePath string, templateParam interface{}, deleteOnExit bool) error {
+	outFilePath, err := h.renderTemplate(templatePath, templateParam)
+	if err != nil {
+		return err
+	}
+
+	if err := h.kubectl("apply", "-f", outFilePath); err != nil {
+		return err
+	}
+
+	if deleteOnExit {
+		h.addCleanupFunc(func() {
+			log.Info("Deleting resources", "file", outFilePath)
+			if err := h.kubectl("delete", "--all", "-f", outFilePath); err != nil {
+				log.Error(err, "Failed to delete resources", "file", outFilePath)
+			}
+		})
+	}
+
+	return nil
+}
+
+func (h *helper) kubectl(command string, args ...string) error {
+	return h.exec(h.kubectlWrapper.Command(command, args...))
+}
+
+func (h *helper) exec(cmd *command.Command) error {
+	ctx, cancelFunc := context.WithTimeout(context.Background(), h.commandTimeout)
+	defer cancelFunc()
+
+	log.V(1).Info("Executing command", "command", cmd)
+	out, err := cmd.Execute(ctx)
+	if err != nil {
+		// suppress the stacktrace when the command fails naturally
+		if _, ok := err.(*exec.ExitError); ok {
+			log.Info("Command returned error code", "command", cmd, "message", err.Error())
+		} else {
+			log.Error(err, "Command execution failed", "command", cmd)
+		}
+
+		fmt.Fprintln(os.Stderr, string(out))
+		return errors.Wrapf(err, "command failed: [%s]", cmd)
+	}
+
+	if log.V(1).Enabled() {
+		fmt.Println(string(out))
+	}
+
+	return nil
+}
+
+func (h *helper) renderTemplate(templatePath string, param interface{}) (string, error) {
+	tmpl, err := template.New(filepath.Base(templatePath)).Funcs(sprig.TxtFuncMap()).ParseFiles(templatePath)
```

Comment:
That is the only place right now. I assumed it would be used more but that doesn't seem to be the case so I am happy to get rid of it.

Reply:
Actually there are no functions in the standard template library to output JSON in any form. I can of course create my own but it feels like having an established library like Sprig might be better because in addition to the JSON functions, there are lots of other functions that could be useful in the future. What do you think?

Reply:
Can we use json.MarshalIndent? https://stackoverflow.com/a/19038873/10628433
No strong opinion though, if you think it's better to keep sprig I'm fine with it.